.github/workflows/dotnet-upgrade.yml

```diff
@@ -0,0 +1,167 @@
+name: "DotNet SDK Upgrade"
+
+on:
+  schedule:
+    - cron: '0 0 * * 1'
+  workflow_dispatch:
+
+env: 
+  MAJOR_MINOR_VERSION: "6.0"
+
+jobs:
+  dotnet-update:
+    runs-on: ubuntu-latest
+    outputs:
+      BRANCH_NAME: ${{ steps.create_branch.outputs.BRANCH_NAME }}
+      LATEST_PATCH_VERSION: ${{ steps.fetch_latest_patch.outputs.DOTNET_LATEST_VERSION }}
+      SHOULD_UPDATE: ${{ steps.fetch_latest_patch.outputs.SHOULD_UPDATE }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Check patch version
+        id: fetch_latest_patch
+        shell: bash
+        run: |
+          latest_patch_version=$(curl -sb -H "Accept: application/json" "https://dotnetcli.blob.core.windows.net/dotnet/Sdk/${{ env.MAJOR_MINOR_VERSION }}/latest.version")
+          current_patch_version=$(jq .sdk.version ./src/global.json | xargs)
+
+          exit_code=0
+          function version { printf "%03d%03d%03d%03d" $(echo "$1" | tr '.' ' '); }
+          
+          [ $(version $current_patch_version) -ge $(version $latest_patch_version) ] && exit_code=1
+
+          echo "DOTNET_CURRENT_VERSION=${current_patch_version}" >> $GITHUB_OUTPUT
+          echo "DOTNET_LATEST_VERSION=${latest_patch_version}" >> $GITHUB_OUTPUT
+          echo "SHOULD_UPDATE=${exit_code}" >> $GITHUB_OUTPUT
```

Comment:
i think we can assume to always to update as long as `DOTNET_CURRENT_VERSION != DOTNET_LATEST_VERSION`?